openapi.yaml

```diff
@@ -1023,6 +1141,372 @@ paths:
+      operationId: getTemporaryRatings
+      description: >
+        Search for temporary ratings.  Will return any temporary ratings that overlap with the 
+        start/end period.  
+
+        Clients SHOULD perform Conditional `GET` using the `If-None-Match` header
+        and the `ETag` of a previous `GET` response.
+      tags:
+        - Temporary Ratings
+      parameters:
+        - $ref: '#/components/parameters/period-start-query'
+        - $ref: '#/components/parameters/period-end'
+        - $ref: '#/components/parameters/monitoring-set-filter'
+        - $ref: '#/components/parameters/segment-filter'
+        - $ref: '#/components/parameters/geographical-region-filter'
+        - $ref: '#/components/parameters/subgeographical-region-filter'        
+        - $ref: '#/components/parameters/naming-scheme'
+        - $ref: '#/components/parameters/limit-units'                 
+      responses: 
+        "200":
+          description: OK
+          content:
+            application/vnd.trolie.temporary-rating-set.v1+json:
+              schema:
+                $ref: '#/components/schemas/temporary-rating-set'
+              example:
+                _links:
+                  self:
+                    href: /temporary-ratings
+                total-count: 1
+                offset: 0
+                _embedded:
+                  temporary-ratings:
+                  - _links:
+                      self: 
+                        href: /temporary-ratings/46f7212b-1633-4c30-ba71-c6e987b2ded7
+                      segment: 
+                        href: /reference-data/network-segments/segmentX
+                        name: segmentX
+                    id: 46f7212b-1633-4c30-ba71-c6e987b2ded7
+                    segment-id: segmentX      
+                    start-time: "2025-07-12T16:00:00-07:00"
+                    end-time:  "2025-07-13T12:00:00-07:00"     
+                    updated-time: "2025-07-08T13:05:43.044267100-07:00"
+                    values:
+                    - type: normal
+                      value:
+                        mva: 160
+                    - type: emergency
+                      value:
+                        mva: 165
+                    - type: loadshed
+                      value:
+                        mva: 170  
+                    temporary-aar-exception: true
+                    temporary-seasonal-rating: false
+                    reason: Free-form text reason.  
+            application/json:
+              schema:
+                $ref: '#/components/schemas/temporary-rating-set'                                          
+          headers:
+            X-Rate-Limit-Limit:
+              $ref: '#/components/headers/X-Rate-Limit-Limit'
+            X-Rate-Limit-Remaining:
+              $ref: '#/components/headers/X-Rate-Limit-Remaining'
+            X-Rate-Limit-Reset:
+              $ref: '#/components/headers/X-Rate-Limit-Reset'
+            X-TROLIE-Limit-Units:
```

Comment:
General feedback on this header:

The media types (input and output / proposals and snapshots) themselves should specify the units, right? So why is this header needed? If a particular TROLIE instance doesn't support the units provided in a submitted proposal because of a business decision, that's not a runtime concern per se. Yes, that TROLIE instance should rejected the proposal, but the sender would have known up front because they know the modeling situation.

Example, MISO only excepts forecasts in MVA for segments, yet only accept MW values for interfaces.

I just don't understand why this header is needed, and it is very unconventional.

Reply:
I think I do have an alternative you'll find acceptable, so please indulge some more context and discussion. I finally get that you'd prefer
```http
GET /forecast
Accept: application/vnd.trolie.forecast-slim+json
X-TROLIE-Limit-Units: MVA
```
to
```http
GET /forecast
Accept: application/vnd.trolie.forecast-slim+json,mva
```

I definitely would not suggest anything that would proliferate media types. My understanding is that the existing schema supports different quantifications (scalars + units) of facility ratings.
![image](https://github.com/ge-grid-software/TROLIE/assets/57670484/dc84bfe3-f588-4419-9609-7e1002d244c2)

MISO may ultimately reject proposals for facility segments that are quantified in units other than MVA, but that is *not* a reason the TROLIE proposal representation can't afford the Ratings Provider the ability to send a proposal quantified in other units. If anything I would be open to *adding* additional options for units in the proposal schema, e.g., MVAR, if someone had the need. In fact, we need a real power only unit, i.e., MW, for interface limits in TROLIE.

I'm not sure how the header helps in submission.  I can see the peering bootstrap concept we've talked about pulling in the peer-TROLIE's profile and learning that it needs to convert to MVA before sending. Is that how you envision this working? Like the LEP client would pre-flight with a `HEAD` request, get a `X-TROLIE-Limit-Units: MVA`, then convert its MW to MVA using the present power factor value?

I'm not sure this is realistic in all cases. MVA to amps requires a known voltage, for example, but that is usually established independently via the voltage solution. Conversion between units also requires agreement on significant digits and/or truncation versus rounding.

Moreover, TROLIE submissions are considered telemetry by MISO. We expect and assume that a telemetered value is the value that the telemetering party is operating against. By extension, they need to be exacting in the value they send: If they operate in amps but MISO operates in MVA, they might use their voltage solution to determine the latter before sending. We could go on, and MISO is aware of the varying opinions on the right units to utilize.

The point is that the representation needs the flexibility to represent any appropriate units for transmission interoperability, but there are engineering decisions that precede the establishment of an exchange, precluding the possibility of a runtime discovery and conversion regime, in general.

In sum, I think the Ratings Obligation profile should specify the acceptable units for proposals per ratings provider as well as the acceptable units for the snapshot representation per monitoring area.

Reply:
I know I was long-winded, so I'll ask for specific feedback on my actual proposal, included below for convenience. I only mentioned the Accept header option to illustrate that I understood the objective. I do not think runtime content negotiation is sufficient to address the objective (which is why I was long-winded).

>I think the Ratings Obligation profile should specify the acceptable units for proposals per ratings provider as well as the acceptable units for the snapshot representation per monitoring area.

What do you think?

Reply:
Oh.  
> I think the Ratings Obligation profile should specify the acceptable units for proposals per ratings provider as well as the acceptable units for the snapshot representation per monitoring area.

Yep, I probably missed that, and fully agree it's simpler than what I've done here in the API.